builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -81,6 +81,15 @@ func resourceCloudStackInstance() *schema.Resource {
 				Set:      schema.HashString,
 			},
 
+			"affinity_group_names": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
```

Comment:
It does not only make sense, but is also required as the params are mutually exclusive according to the docs: http://cloudstack.apache.org/docs/api/apidocs-4.8/root_admin/updateVMAffinityGroup.html